### PR TITLE
rosie: unstable-2020-01-11 -> 1.3.0

### PR DESCRIPTION
**Pull request by blin**
###### Description of changes

unstable-2020-01-11 seems to be between 1.2.1 and 1.2.2 , based on https://gitlab.com/rosie-pattern-language/rosie/-/blob/670e9027563609ba2ea31e14e2621a1302742795/CHANGELOG . There has been no breaking changes since, according to https://gitlab.com/rosie-pattern-language/rosie/-/blob/v1.3.0/CHANGELOG?ref_type=tags .

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
Needs patches on darwin:

```nix
  preInstall = lib.optionalString stdenv.isDarwin ''
    install_name_tool -add_rpath $out/lib build/bin/rosie
    install_name_tool -id $out/lib/librosie.dylib build/lib/librosie.dylib
  '';
```

**Comment by blin** [comment]
>Needs patches on darwin:

Done. 

I don't know how to test this on darwin, please let me know if you have any suggestions.

**Comment by wegank** [comment]
@ofborg build rosie rosie.passthru.tests

File: pkgs/tools/text/rosie/default.nix [hunk 1]
```diff
@@ -23,17 +23,31 @@ stdenv.mkDerivation rec {
...
+    # and so needs to be patched in preConfigure.
+    patchShebangs rosie
     # Part of the same Makefile target which calls git to update submodules
     ln -s src submodules/lua/include
+    # ldconfig is irrelevant, disable it inside `make installforce`.
+    sed -iE 's/ldconfig/echo skippin ldconfig/' Makefile
+    sed -iE '/ld.so.conf.d/d' Makefile
+  '';
+
+  preInstall = ''
```
Comment (wegank) [inline pkgs/tools/text/rosie/default.nix:36]:
```suggestion
  preInstall = lib.optionalString stdenv.isDarwin ''
```

Comment (blin) [reply pkgs/tools/text/rosie/default.nix]: Good catch, bad copy-paste on my part.

**Comment by wegank** [comment]
Please squash the commits, thanks!

**Comment by blin** [comment]
>Please squash the commits, thanks!

Done.

**Pull request merged**